flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala

```diff
@@ -2008,6 +2008,33 @@ class CodeGenerator(
   }
 
   /**
+    * Adds a reusable [[java.util.Random]] to the member area of the generated [[Function]].
+    *
+    * The seed parameter must be a literal/constant expression.
+    *
+    * @return member variable term
+    */
+  def addReusableDateFormatter(format: GeneratedExpression): String = {
+    val fieldTerm = newName("dateFormatter")
+
+    val field =
+      s"""
+         |transient org.joda.time.format.DateTimeFormatter $fieldTerm;
+         |""".stripMargin
+    reusableMemberStatements.add(field)
+
+    val fieldInit =
+      s"""
+         |${format.code}
+         |$fieldTerm = org.apache.flink.table.runtime.functions.
+         |DateTimeFunctions$$.MODULE$$.createDateTimeFormatter(${format.resultTerm});
```

Comment:
This is because the runtime ``DateTimeFunctions`` has been ported to Scala. This is the syntax to access the function.